presto-main/src/main/java/com/facebook/presto/dispatcher/DispatchManager.java

```diff
@@ -239,6 +246,35 @@ private <C> void createQueryInternal(QueryId queryId, String slug, int retryCoun
         }
     }
 
+    private Session checkPermissions(Session session)
```

Comment:
This does not seem correct. Since we want this new API as a replcement for the checkDeletegation, shouldn't this logic only be present in the PrismSystemAccessControl?

Reply:
The api is defined there; but it will be called here.

Reply:
I don't think we should have this API here.